src/main/java/com/esri/arcgisruntime/toolkit/skins/FloorFilterSkin.java

```diff
@@ -0,0 +1,706 @@
+   * @since 100.14.0
+   */
+  public FloorFilterSkin(FloorFilter control) {
+    super(control);
+
+    // add a listener on the floor manager property
+    // the UI will only display and be configured if a floor manager is loaded successfully in the control
+    control.floorManagerProperty().addListener(observable -> handleFloorManagerChanged());
+
+    // set boolean property to true if the geo view in the control is a scene view
+    isSceneViewProperty.set(getSkinnable().getGeoView() instanceof SceneView);
+
+    // add listeners to selected site/facility/level properties
+    control.selectedSiteProperty().addListener(observable -> handleSelectedSiteChanged());
+    control.selectedFacilityProperty().addListener(observable -> handleSelectedFacilityChanged());
+    control.selectedLevelProperty().addListener(observable -> handleSelectedLevelChanged());
+    selectedSiteProperty.bind(control.selectedSiteProperty());
+    selectedFacilityProperty.bind(control.selectedFacilityProperty());
+
+    // handle the all sites property
+    isAllSitesProperty.addListener(observable -> handleIsAllSitesPropertyChanged());
+    // handle the expansion of the titled panes
+    sitesTitledPane.expandedProperty().addListener(observable -> handleTitledPaneExpanded(sitesTitledPane));
+    facilitiesTitledPane.expandedProperty().addListener(observable -> handleTitledPaneExpanded(facilitiesTitledPane));
+    // handle the collapsed view
+    isCollapsedProperty.addListener(observable -> handleIsCollapsedPropertyChanged());
+  }
+
+  /**
+   * Handles changes to the floor manager property value. This configures properties that relate to how the UI should
+   * display depending on the data, and triggers the UI to draw if it is the first time a floor manager has been set.
+   * If the floor manager changes as a result of the control.refresh() method being called, the data will be configured
+   * within the existing UI.
+   *
+   * @since 100.14.0
+   */
+  private void handleFloorManagerChanged() {
+    if (getSkinnable().getFloorManager() != null) {
+      // if the floor filter has not been drawn already, set up the UI elements
+      if (floorFilterWrapper.getChildren().isEmpty()) {
+        setupUI();
+      }
+
+      // if there are no sites, don't show the sites browser
+      if (getSkinnable().getSites().isEmpty()) {
+        isShowSitesProperty.set(false);
+      } else if (getSkinnable().getSites().size() == 1) {
+        // if there is 1 site, set it as the selected site and don't show the sites browser
+        getSkinnable().setSelectedSite(getSkinnable().getSites().get(0));
+        isShowSitesProperty.set(false);
+      } else {
+        // if there are multiple sites, show the sites browser and set the data to the sites list
+        isShowSitesProperty.set(true);
+        sites.setAll(getSkinnable().getSites());
+        // sort the sites by name
+        sites.sort(Comparator.comparing(FloorSite::getName));
```

Comment:
The logic is that if there is only 1 site or facility, we don't need to let the user browse so we auto-select it, and in the case of the sites we don't even show the site browser. If there are multiple sites we want the user to make the choice first rather than having one auto-selected. So I think I'll need to leave this logic based on the design discussions with the indoors team.